A4A > Migrations: Implement the self migration tool

yashwin opened the pull request:
Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/1477
Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/1478

## Proposed Changes

This PR

- Implements a self migration tool for A4A. 
- Adds TaskSteps component that stores the state using session storage.

Note: The actual steps will be added in another PR. These are just dummy steps. 

## Why are these changes being made?

* To allow users to track the progress of self migrating a site. 

@madebynoam: Do you think we should collapse the completed steps when the user visits this page again?

## Testing Instructions

* Open the A4A live link 
* Go to Migrations: Overview: Click the `Migrate your site` button > Select the `Self migrate to WordPress.com` option > Verify the below is displayed. Play around with the tool. Clicking on `Mark as done` should set the step as completed and `Reset task` should reset the status. Refreshing the page should restore the previous state as we are storing it in session storage. Clicking on the `Reset task` should reset the state to default. 

<img width="1728" alt="Screenshot 2024-11-15 at 3 30 29 PM" src="https://github.com/user-attachments/assets/70f3e240-d6fd-499f-9265-6e742be4c9ff">

* Repeat the same for the `Self migrate to Pressable` option

<img width="1728" alt="Screenshot 2024-11-15 at 3 30 33 PM" src="https://github.com/user-attachments/assets/81934d4f-b221-48c4-ad32-92066e1c37db">

* Verify it looks good across different screen sizes.

<img width="768" alt="Screenshot 2024-11-15 at 3 30 58 PM" src="https://github.com/user-attachments/assets/fc85da9b-9528-460c-af55-94da8dc21f67">

<img width="372" alt="Screenshot 2024-11-15 at 3 33 25 PM" src="https://github.com/user-attachments/assets/4e91268b-1af6-4ac7-873b-f9e43f28ea85">


When a few steps are completed:

<img width="1728" alt="Screenshot 2024-11-15 at 3 30 24 PM" src="https://github.com/user-attachments/assets/e8555015-45bc-404f-b873-dfa9a9ff8953">


When all the steps are collapsed: 

<img width="1728" alt="Screenshot 2024-11-15 at 3 33 47 PM" src="https://github.com/user-attachments/assets/d3cbd766-fb07-4fcd-b79c-f09e4b05ce64">


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

yashwin committed:
Implement self migration tool

madebynoam commented:
Lgtm, @yashwin!
Noticed a bug with the breadcrumb, where clicking "Overview" doesn't navigate to the Overview page.
<img width="475" alt="image" src="https://github.com/user-attachments/assets/22c6bf15-c890-48fe-b173-2eaa2e3afbad">

yashwin committed:
address PR comment

yashwin committed:
remove translation

yashwin commented:
Thanks for the review, @madebynoam!

I have addressed your comment and updated it now. We not also have an option to reset the individual step.

<img width="1728" alt="Screenshot 2024-11-15 at 3 30 24 PM" src="https://github.com/user-attachments/assets/f834c049-8ef8-45b8-bac3-95884ef0c141">

jkguidaven commented:
@yashwin @madebynoam When the card expands, it shifts downward slightly. Is this intentional?

https://github.com/user-attachments/assets/0b52a70f-b3c7-4517-85b7-41d901b50d9a

jkguidaven commented on client/a8c-for-agencies/components/task-steps/index.tsx:
```diff
@@ -0,0 +1,120 @@
+			clickableHeader
+			summary={ false }
+		>
+			<>
```
Is the extra Fragment here necessary?

yashwin replied:
Good catch. Removed it here: https://github.com/Automattic/wp-calypso/pull/96443/commits/c67444553b6836c9662c244c1f48c364bf621126

jkguidaven commented on client/a8c-for-agencies/components/task-steps/style.scss:
```diff
@@ -0,0 +1,120 @@
+@import "@wordpress/base-styles/breakpoints";
+@import "@wordpress/base-styles/mixins";
+
+.task-steps {
```
Following the [Calypso CSS guidelines](https://github.com/Automattic/wp-calypso/blob/trunk/docs/coding-guidelines/css.md#nesting), we should aim for a nesting level of two. It may not be necessary to nest everything under `.task-steps`, as all child components are already specific (`.task-steps__*`).

yashwin replied:
Totally agree. Fixed it here: https://github.com/Automattic/wp-calypso/pull/96443/commits/c67444553b6836c9662c244c1f48c364bf621126

jkguidaven reviewed (commented):
Overall it looks good! Nice work!

I left few comments for improvement.

yashwin committed:
minor UI fixes

yashwin commented:
> Overall it looks good! Nice work!

Thanks for the review, @jkguidaven!

> @yashwin @madebynoam When the card expands, it shifts downward slightly. Is this intentional?

No, this is not intentional. Good catch. I have fixed it now: https://github.com/Automattic/wp-calypso/pull/96443/commits/c67444553b6836c9662c244c1f48c364bf621126

> I left few comments for improvement.

I have addressed all your comments. Could you please take another look?

jkguidaven reviewed (approved):
Thank you for addressing my comments LGTM 👍

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17007681).
